src/middlewared/middlewared/plugins/cluster_linux/ctdb_job.py

```diff
@@ -0,0 +1,136 @@
+from middlewared.service import Service, private, job, periodic
+from middlewared.service_exception import CallError
+from middlewared.plugins.cluster_linux.utils import CTDBConfig
+
+import time
+import os
+import errno
+import enum
+import asyncio
+
+class CLStatus(enum.Enum):
+    ACTIVE = enum.auto()
+    EXPIRED = enum.auto()
+
+
+class ClusterJob(Service):
+    class Config:
+        private = True
+
+    async def list(self):
+        """
+        Return queue indexed by pnn
+        """
+        cl_jobs = await self.middleware.call(
+            'clustercache.query',
+            [('key', '^', 'CLJOB_')],
+        )
+        output = {}
+        now = time.clock_gettime(time.CLOCK_REALTIME)
+
+        for i in cl_jobs:
+            pnn = int(i['key'].rsplit('_', 1)[1])
+            if i['timeout'] and now > i['timeout']:
+                status = CLStatus.EXPIRED
+            else:
+                status = CLStatus.ACTIVE
+
+            entry = {
+                'key': i['key'],
+                'pnn': pnn,
+                'status': status.name,
+                'method': i['value']['method'],
+                'payload': i['value'].get('data', None)
+            }
+
+            if pnn not in output:
+                output[pnn] = [entry]
+            else:
+                output[pnn].append(entry)
+
+        return output
+
+    @private
+    @periodic(3600)
+    @job(lock="queue_lock")
+    async def process_queue(self, job):
+        gl_enabled = (await self.middleware.call('service.query', [('service', '=', 'glusterd')], {'get': True}))['enable']
+        if not gl_enabled:
+            return
+
+        node = (await self.middleware.call('ctdb.general.status', {'all_nodes': False}))[0]
```

Comment:
This will raise a `CallError` if the ctdb service isn't running so not sure if you want that or if you want to add another check up above.

Reply:
I think raising a CallError in this case is fine.